app/src/main/kotlin/cz/covid19cz/app/ui/confirm/DeleteUserFragment.kt

```diff
@@ -2,20 +2,20 @@ package cz.covid19cz.app.ui.confirm
 
 import cz.covid19cz.app.R
 import cz.covid19cz.app.service.CovidService
+import cz.covid19cz.app.ui.notifications.CovidNotificationManager
 import cz.covid19cz.app.utils.Auth
 
 class DeleteUserFragment : ConfirmationFragment() {
     override val description by lazy { getString(R.string.delete_user_desc, Auth.getPhoneNumber())}
     override val buttonTextRes = R.string.delete_registration
     override fun confirmedClicked() {
-        context?.let {
-            it.startService(CovidService.stopService(it))
-        }
         viewModel.deleteUser()
     }
 
     override fun doWhenFinished() {
+        context?.let {
+            it.startService(CovidService.stopService(it, true))
+        }
```

Comment:
```suggestion
       requireContext().startService(CovidService.stopService(it, true))   
```

Reply:
Dá se tomu ViewModelu věřit, že nikdy nedoručí tu asynchronní zprávu tomu fragmentu v momentě, kdy ten fragment už je odpojený od aktivity/zrušený?

Reply:
Hele teoreticky by asi mohl byt od aktivity odpojeny. Mam radsi kdyz to pak crashne rovnou, nez aby aplikace skoncila v nejakem divnem mezistavu. Ale je to fakt detail.

Reply:
Je správné crashnout appku, když je chyba v kódu, ale v Androidu to je spíše "chyba" uživatele nebo OS, že se zmáčkne tlačítko zpět ve špatnou milisekundu :D Zastavení služby je v tomhle případě detail, tak bych to nechal tak.